### ISSUE TITLE
Kitchen converge fails to copy cookbooks over winrm in certain cases

### ISSUE THREAD

**Issue by jaym** [test-kitchen/test-kitchen#1275]
## Description
We're failing to copy cookbooks over correctly in cases where the the temporary directory has a short name in it. This is due to in issue in winrm-fs where it's using glob on the directory we pass it, and then assuming all paths start with that directory. Since glob is expanding the short name, that's not true (at least form a string comparison stance). This is fallout from out 1.1.17 release where we stopped using glob and thus stopped expanding those short names.

A pry session showing that:
```
    118: def produce_zip_entries(zos)
    119:   require 'pry';binding.pry
    120:   entries.each do |entry|
    121:     entry_path = entry.sub(/#{dir}\//i, '')
    122:     logger.debug "+++ Adding #{entry_path}"
    123:     binding.pry
 => 124:     zos.put_next_entry(
    125:       zip_entry(entry_path),
    126:       nil, nil, ::Zip::Entry::DEFLATED, Zlib::BEST_COMPRESSION)
    127:     entry.open('rb') { |src| IO.copy_stream(src, zos) }
    128:   end
    129:   binding.pry
    130:   logger.debug '=== All files added.'
    131: end

[1] pry(#<WinRM::FS::Core::TmpZip>)> entry_path
=> #<Pathname:C:/Users/Administrator/AppData/Local/Temp/2/default-windows-2012r2-sandbox-20170914-2440-1pb86hg/cookbooks/workstation/README.md>

[2] pry(#<WinRM::FS::Core::TmpZip>)> dir
=> #<Pathname:C:/Users/ADMINI~1/AppData/Local/Temp/2/default-windows-2012r2-sandbox-20170914-2440-1pb86hg/cookbooks>
```

**Comment by manheraz** [comment]
Hi there!

Is this bug related to the issue explained in this message?

[Error Resolving Cookbooks for Run List](https://discourse.chef.io/t/resolved-error-resolving-cookbooks-for-run-list/11376)

The command `kitchen converge` doesn't work in my computer since chefdk 2.0.28 (I have tried chefdk 2.2.1 and chefdk 2.3.1)

As explained in the mailing list message, the cookbooks folder is empty in the virtual machine (Windows Server 2016 Standard with GUI evaluation in my case, not 2012R2)

My development OS is Windows 10 1511, and I'm using the hyperv driver.

This is the complete .kitchen.yml file:

```yaml
---
driver:
  name: hyperv
  parent_vhd_folder: 'X:\Hyper-V\Windows Server 2016\Virtual Hard Disks'
  parent_vhd_name: tkWS2016.vhdx
  vm_switch: ExternalSwitch
  processor_count: 2
  memory_startup_bytes: 2GB
  dynamic_memory: false
  vm_generation: 2

provisioner:
  name: chef_zero
  # You may wish to disable always updating cookbooks in CI or other testing environments.
  # For example:
  #   always_update_cookbooks: <%= !ENV['CI'] %>
  always_update_cookbooks: true

  # Settings to handle reboots
  retry_on_exit_code:
    - 35 # 35 is the exit code signaling that the node is rebooting
  max_retries: 5
  wait_for_retry: 60

transport:
  name: winrm
  username: Administrator
  password: *************
  elevated: true

verifier:
  name: pester

platforms:
  - name: windows-server-2016-std

suites:
  - name: default
    run_list:
      - recipe[mdt::default]
    attributes:
```

Thank you in advance.

**Comment by jaym** [comment]
I wouldn't be surprised if it was. You can try working around it by creating a folder `c:\tmp` and setting `TMP` and `TEMP` (I cant remember which one actually gets used) to `"c:\tmp"`

**Comment by manheraz** [comment]
Thanks!

I set the **TEMP** user environment variable to `C:\tmp` and `kitchen converge` worked.

I haven't changed the **TMP** environment variable (it stills point to `%USERPROFILE%\AppData\Local\Temp`)

This bug hit me in my work computer, where I have a long Windows username (> 15 characters). In my personal workstation, where I'm using a short Windows username, everything is working without any workaround.

This bug is still present in ChefDK 2.3.4.

**Comment by jaym** [comment]
Bug should be fixed in the next release of chefdk/test-kitchen. Seems it's the same bug

**Comment by cheeseplus** [comment]
This was resolved in [1.18.0](https://github.com/test-kitchen/test-kitchen/blob/master/CHANGELOG.md#v1180-2017-09-28)
